screens/Menu/ShuttleBus.js

```diff
@@ -1,32 +1,245 @@
+     * shuttle bus stops
+     * @param {*} nextStop 
+     */
+    const calculateTimeDifference = (nextStop) => {
+        var diffHours = nextStop.hour - currentTime.Hour;
+        var diffMinutes = nextStop.minutes - currentTime.Minutes;
+        if (diffHours >= 1 && Math.abs(diffMinutes) < 10) {
+            return ({
+                Hours: diffHours + "h",
+                Minutes: "0" + Math.abs(diffMinutes) + "min"
+            });
+        }
+        else if (diffHours < 1) {
+            return ({
+                Hours: "",
+                Minutes: Math.abs(diffMinutes) + " min"
+            });
+        }
+        else if (Math.abs(diffMinutes) < 1) {
+            return ({
+                Hours: diffHours + " h",
+                Minutes: ""
+            });
+        }
+        else {
+            return ({
+                Hours: diffHours + "h",
+                Minutes: Math.abs(diffMinutes) + "min"
+            });
+        }
+
+    };
+
+    /**
+     * Function that returns an object containing the id, timeDifference as well as the time of the next buses on schedule
+     * @param {*} list 
+     */
+    const fetchNextStops = (list) => {
+        var nextStops = [];
+        var results = [];
+        let index = 0;
+        for (var key in list) {
+            if (list[key].hour >= currentTime.Hour && list[key].minutes >= currentTime.Minutes) {
+                nextStops = list.slice(index, list.length - 1);
+                break;
+            }
+            index++;
+        }
+        if (nextStops.length > 0) {
+            results = nextStops.map((element, index) => {
+                return ({
+                    id: index.toString(),
+                    timeDifference: calculateTimeDifference(element),
+                    hour: element.hour,
+                    minutes: element.minutes
+                });
+            });
+        }
+        return results;
+    };
+
+    /**
+     * Function that sorts through the shuttle bus schedule and compares the time/day to the current time/day
+     */
+    const getNextStops = () => {
+        var scheduleTimesSGW = [];
+        var scheduleTimesLoyola = [];
+        const sgwCampus = "SGW";
+        const loyolaCampus = "Loyola";
+        if (currentDay && weekDays.includes(currentDay.toString()) && !friday.includes(currentDay.toString()) && !weekends.includes(currentDay.toString())) {
+            scheduleTimesSGW = getShuttleBusTimes[sgwCampus].MondayToThursday;
+            scheduleTimesLoyola = getShuttleBusTimes[loyolaCampus].MondayToThursday;
+        }
+        else if (currentDay && friday.includes(currentDay.toString())) {
+            scheduleTimesSGW = getShuttleBusTimes[sgwCampus].Friday;
+            scheduleTimesLoyola = getShuttleBusTimes[loyolaCampus].Friday;
+        }
+        else {
+            return alert("There is no shuttle bus on weekends. Please check back during the week!");
```

Comment:
I prefer a message written on the tabs (same on each campus) that let the user know that there are no stops for weekends... Because when nothing is returns, it's just nothing... Also, for when there are no more busses (like today its tuesday and the last is at 10:30PM: maybe say, no more stops for the day, check again tomorrow.